### DISCUSSION TITLE
Best Inverter/Battery API

### DISCUSSION THREAD

**Discussion by elkgeek** [amberelectric/public-api#219] [Q&A]
I'm looking to buy a new PV system. I want to be able to program it to charge/discharge from an API. A decent API/rest/modbus/whatever is a key criteria in my choice.

I'm an experienced dev, learning a new API isn't an issue.

What are you using and how are you finding programming your system?

thanks

**Comment by konstar** [comment]
I chose a SolarEdge Inverter and Battery, almost 2 years ago, and it has been excellent.

Have relied on Home Assistant and the solaredge-modbus-multi integration to be able to do everything I've needed. You can control the inverter and battery immediately, for any situation.

I ran with SmartShift for a few months and switched it off after I was confident enough to automate control myself.

It's worked so well, I'm thinking of getting another SolarEdge battery.

**Reply by chamaelion** [reply]
Hi @konstar - I too have a SolarEdge Inverter and Battery and am loving it, probably for the same reasons as you :)  I've been using the feeds from [solaredge-modbus-multi](https://github.com/WillCodeForCats/solaredge-modbus-multi) for so much automation, allowing me to fine tune much of my power use via HA automations.

However, after recently joining Amber I've wanted to move from just leveraging the sensor feeds to (as you have) allow HA to control the inverter based on the feed from the Amber API.  Reacting to the API's no problem - I've sanity checked this by setting up smart lights to come on and change colour based on input from the API such as feed_in_price_descriptor - _but I can't get my commands to flow through from HA to the inverter_.

I've followed the integration's [Power Control](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Power-Control-Options-%E2%80%90-Configuration) section, as well as double-checked my setup from the [Configuration](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration#integration-configuration-options) page, but when I try to (for example) set **Storage Control Mode** to **_Remote Control,_** with the intent of then setting **Storage Command** to **_Discharge Maximum for Export_**, HA consistently throws an error - "_failed to perform the action {blah}. Connection to inverter ID 1 failed._"

I'm almost certain that this is because I haven't properly configured the Inverter to accept remote control commands, and I've made sure that **Advanced Power Control** in HA is set to **_On_**.  [Other resources](https://github.com/binsentsu/home-assistant-solaredge-modbus/blob/master/README.md) (specifically at the last heading **Enabling Power Control on SolarEdge Inverter**) have suggested connecting to the inverter via the maintenance panel locally at 172.16.0.1 and setting **Reactive Power Mode** to **_RRCR_** - this unfortunately has had no effect, so for now I've set **Reactive Power Mode** back to the default.  It also mentions setting **Advanced Power Control** to **_Enable_**, _but I cannot find that setting in the inverter maintenance panel after an exhaustive search_.

Any chance you could assist in either correcting my assumptions and / or pointing me in the right direction please?  Thanks in advance!

**Comment by craigcurtin-dev** [comment]
I have a combination of systems - 3 x single phase GoodWe AC coupled batery inverters that controls my 40Kwh of battery across 3 phases. As value for money they are almost unbeatable and they support LV batteries so that is a definite win. I control them through Hardwired Modbus to their EMS port thriugh Node Red.

I also have two x PV inverters - which are Solis units and they have been rock solid also - again Modbus through their external RS485 connectors. I can control these to limit grid output manually.

All of my house measurements are done using IOTAWATTS

Craig

**Comment by MadogMedia** [comment]
I consider LUX ON-Grid Inverters are the Best !

As they have the best Programming Portal, with multiple Charge & Three programmable Forced Discharge Time Slots, that suits the Amber High FIT & Low Power prices.
